massa-execution-worker/src/execution.rs

```diff
@@ -1354,4 +1406,95 @@ impl ExecutionState {
     pub fn get_address_future_deferred_credits(&self, address: &Address) -> BTreeMap<Slot, Amount> {
         context_guard!(self).get_address_future_deferred_credits(address, self.config.thread_count)
     }
+
+    /// Get the execution statuses of both speculative and final executions
+    ///
+    /// # Return
+    ///
+    /// * A tuple of hashmaps with:
+    /// * first the statuses for speculative executions
+    /// * second the statuses for final executions
+    pub fn get_op_exec_status(&self) -> (HashMap<OperationId, bool>, HashMap<OperationId, bool>) {
+        (
+            self.active_history.read().get_op_exec_status(),
+            self.final_state.read().executed_ops.op_exec_status.clone(),
```

Comment:
Either comment that the returned values are deep copies, thus removing synchronisations on the data, or provide the data in the form of a read-guard so as to retain the syncro, leaving it up to the function caller to opt-out of the syncronisation

it might be preferable to return a read-guard here? The returned hash-maps are deep-copies, removing any syncro of the data.

Reply:
this implementation is incorrect on multiple levels:
* the returned final and active values can be desync
* the active one does not take the final one into account as fallback if no info found
* in the active one we take the latest inclusion of the operation as reference but we shouldn't

More generally the whole executed ops changes need a full review because there seem to be multiple problems

Reply:
this needs a followup issue

Reply:
#3704